### PR TITLE
Support tilde token for concatenation of string literals

### PR DESCRIPTION
**Pull request by wamuir**
My read of the RELAX NG Compact Syntax spec indicates that two string literals should be concatenated when immediately surrounding a tilde.  This is not currently supported by rnc2rng but could be useful.  As one example, Appendix B to RFC 4278 (ATOM syndication) utilizes several tildes and concatenation would be expected.  PR to include support in parser.py and add a simple test.

### CODE REVIEW

File: rnc2rng/parser.py [hunk 1]
```diff
@@ -50,7 +50,8 @@ def lex(src):
     'AMP', 'CNAME', 'COMBINE', 'COMMA', 'DOCUMENTATION', 'EQUAL', 'ID',
     'LBRACE', 'LBRACKET', 'LPAREN', 'LIST', 'LITERAL', 'MINUS', 'MIXED',
     'PLUS', 'PIPE', 'QMARK', 'RBRACE', 'RBRACKET', 'RPAREN', 'STAR', 'TILDE',
-] + [s.upper() for s in KEYWORDS])
+] + [s.upper() for s in KEYWORDS], precedence=[("left", ['TILDE'])])
```
Comment (djc) [inline rnc2rng/parser.py]: Is the explicit precedence here actually needed?

File: rnc2rng/parser.py [hunk 2]
```diff
@@ -78,6 +79,15 @@ def __repr__(self):
 def start(s, p):
     return Node('ROOT', None, p[0] + p[1])
 
+@pg.production('strlit : LITERAL')
+def primary_literal(s, p): # from datatypeValue
+    return p[0]
+
+@pg.production('strlit : strlit TILDE strlit')
+def primary_concat(s, p):
+    p[0].value = p[0].value + p[2].value
```
Comment (djc) [inline rnc2rng/parser.py]: Doing `+=` here would be slightly nicer, I think, do you agree?

**Review by djc — CHANGES_REQUESTED** [review]
Thanks for the PR! This should definitely support `~`, I just hadn't come across anything using it yet (and I may have missed it when reading the spec). I requested a few minor changes, are you up for making those? Otherwise I can just fix things up myself.

Additionally, I'd prefer to fold the first and last commits into the relevant other commits so I have nicer commit history, but I can take care of that easily if you don't want to get into that.

File: rnc2rng/parser.py (hunk 2 above)
Comment (wamuir) [reply rnc2rng/parser.py]: For sure!

File: rnc2rng/parser.py (hunk 1 above)
Comment (wamuir) [reply rnc2rng/parser.py]: Yes, I think so.  Although I'm not sure of a better way to avoid a conflict.

**Comment by wamuir** [comment]
Great, thanks!  I made the operator change (left a comment on the first) and squashed commits.

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Merged! I've also pushed a new release to PyPI, thanks.

I'd like to add you to the AUTHORS file if you don't mind passing me a real email address (or make a PR yourself).

**Comment by wamuir** [comment]
Sure!  Address is wamuir@gmail.com

**Comment by djc** [comment]
Added in d96073e.

**Pull request merged**